navigator_helpers/tasks/evaluation/evaluation.py

```diff
@@ -0,0 +1,764 @@
+            # Count datetime fields as "Other"
+            try:
+                # pd.to_datetime(non_na_data)
+                pd.to_datetime(non_na_data, format="%Y-%m-%d")
+                return "Other"
+            except:
+                pass
+            return "Text"
+
+        return "Other"
+
+    def _get_tfidf_vectors(self, column: Series) -> np.ndarray:
+        vectorizer = TfidfVectorizer().fit_transform(column)
+        return vectorizer.toarray()
+
+    def _get_llm_as_a_judge_prompt(self, natural_language: str, code: str) -> str:
+        rubric = llm_as_a_judge_prompts.general_response_quality_rubric
+        prompt = rubric(
+            natural_language=natural_language,
+            code=code,
+        )
+        return prompt
+
+    def _eval_response_with_llm_as_a_judge(
+        self,
+        natural_language: str,
+        code: str,
+        **kwargs,
+    ) -> dict:
+        prompt = self._get_llm_as_a_judge_prompt(
+            natural_language=natural_language, code=code, **kwargs
+        )
+        scores = {}
+        response = utils.parse_json_str(self.llm_suite.judge_generate(prompt)) or {}
+        for k, v in response.items():
+            scores[f"{k}_score"] = v["score"]
+            scores[f"{k}_reason"] = v["reasoning"]
+        return scores
+
+    def row_uniqueness(self):
+        """
+        Evaluation test for brute force row uniqueness & semantic uniqueness using TF-IDF and cosine similarity.
+        Returns:
+            - Percentage of rows that are unique
+            - Percentage of rows that are semantically unique (fuzzy match)
+            - IDs of rows that are not unique
+            - IDs of rows that are not semantically unique
+        Example for semantic duplicates:
+            - Prompt 1 : "How to implement k-means clustering in Python?"
+            - Prompt 2 : "Can you implement k-means clustering in Python?"
+        """
+        # TODO: Try out https://github.com/dleemiller/WordLlama for semantic uniqueness
+
+        total_rows = len(self.dataset)
+
+        # Brute Force row uniqueness by detecting duplicates
+        unique_rows = self.dataset.drop_duplicates()
+        non_unique_ids = self.dataset[self.dataset.duplicated()].index.tolist()
+        percent_unique = len(unique_rows) / total_rows * 100
+
+        # Semantic uniqueness using cosine similarity and TF-IDF
+        text_columns = self.dataset.select_dtypes(include=[object])
+        # text_columns = self.dataset.select_dtypes(include=[object, 'string'])
+        concatenated_text = text_columns.apply(
+            lambda row: " ".join(row.values.astype(str)), axis=1
+        )
+        overall_vectors = self._get_tfidf_vectors(concatenated_text)
+        cosine_sim = cosine_similarity(overall_vectors)
+        np.fill_diagonal(cosine_sim, 0)
+        cosine_sim[np.tril_indices(cosine_sim.shape[0])] = 0
+        non_semantically_unique_ids = np.where((cosine_sim > 0.85) & (cosine_sim < 1.0))
```

Comment:
A comment explaining how the two values of 0.85 and 0.1 were driven would be helpful.